src/Type/Php/FilterVarDynamicReturnTypeExtension.php

```diff
@@ -22,12 +24,19 @@ class FilterVarDynamicReturnTypeExtension implements DynamicFunctionReturnTypeEx
 	/** @var array<string, Type> */
 	private $filterTypesHashMaps;
 
+	/** @var array<string, Type> */
+	private $nullableTypes;
+
 	public function __construct()
 	{
 		$booleanType = new BooleanType();
+		$nullableBooleanType = new UnionType([new BooleanType(), new NullType()]);
 		$floatOrFalseType = new UnionType([new FloatType(), new ConstantBooleanType(false)]);
+		$nullableFloatType = new UnionType([new FloatType(), new NullType()]);
 		$intOrFalseType = new UnionType([new IntegerType(), new ConstantBooleanType(false)]);
+		$nullableIntType = new UnionType([new IntegerType(), new NullType()]);
 		$stringOrFalseType = new UnionType([new StringType(), new ConstantBooleanType(false)]);
+		$nullableStringType = new UnionType([new StringType(), new NullType()]);
```

Comment:
Instances of same types (like `new StringType()`) should be reused here from the same variables.